### PR TITLE
Add checkoutLink and defaultLang to GreenField invoice

### PR DESCRIPTION
**Pull request by NicolasDorier**

### CODE REVIEW

File: BTCPayServer.Client/Models/CreateInvoiceRequest.cs [hunk 1]
```diff
@@ -31,9 +31,9 @@ public class CheckoutOptions
             public TimeSpan? Monitoring { get; set; }
 
             public double? PaymentTolerance { get; set; }
-
             [JsonProperty("redirectURL")]
             public string RedirectURL { get; set; }
+            public string DefaultLang { get; set; }
```
Comment (Kukks) [inline BTCPayServer.Client/Models/CreateInvoiceRequest.cs:36]: can we name this better now?
`
```suggestion
            public string DefaultLanguage { get; set; }
```

File: BTCPayServer.Client/Models/InvoiceData.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@ namespace BTCPayServer.Client.Models
     public class InvoiceData : CreateInvoiceRequest
     {
         public string Id { get; set; }
+        public string CheckoutLink { get; set; }
```
Comment (Kukks) [inline BTCPayServer.Client/Models/InvoiceData.cs:10]: ugh I would really prefer we don't do this. It is very opinionated and depends a lot on the calling environment (multiple domains, onion, reverse proxy, ssh tunnelling, etc)

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer.Client/Models/CreateInvoiceRequest.cs (hunk 1 above)
Comment (NicolasDorier) [reply BTCPayServer.Client/Models/CreateInvoiceRequest.cs:36]: Actually I started doing that, but "Lang" is commonly used. No strong pref though.

File: BTCPayServer.Client/Models/InvoiceData.cs (hunk 2 above)
Comment (NicolasDorier) [reply BTCPayServer.Client/Models/InvoiceData.cs:10]: No, we fill that out from the HttpRequest which gives the link adapted. (ie. if you query on onion, you get onion link)

People should not have to build the URL by themselves.

**Comment by NicolasDorier** [comment]
renamed

**Pull request merged**